fix/package-dts-error

Pickra opened the pull request:
# Description
When upgrading to `Remix-Dev-Tools` v2, I noticed a type error in the package. i was getting the error when running `tsc` on my repo which consumes `Remix-Dev-Tools`. This fixes it, but I'm not sure if you're cool with my solution.

<img width="1401" alt="Screenshot 2023-08-09 at 5 49 22 PM" src="https://github.com/Code-Forge-Net/Remix-Dev-Tools/assets/5226549/72788546-4ec1-4581-be0f-b00f3df7697d">


## Type of change: Bug fix
Move `Tab` type export to top level. This is a minor change that doesn't affect the current `dist` build folder structure or naming convention. 

# How Has This Been Tested?

I ran `npm run build` in remix-development-tools, the below `dist` error exists in `main` but not this branch.
```ts
dist/RemixDevTools/RemixDevTools.d.ts:1:21 - error TS2307: Cannot find module './tabs' or its corresponding type declarations.

1 import { Tab } from "./tabs";
                      ~~~~~~~~
```

# Checklist:

- [x] My code follows the guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas; NOT APPLICABLE
- [ ] I have made corresponding changes to the documentation; NOT APPLICABLE
- [x] My changes generate no new warnings or errors
- [ ] I have added tests that prove my fix is effective or that my feature works; NOT APPLICABLE
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules; NOT APPLICABLE

AlemTuzlak commented on src/RemixDevTools/index.ts:
```diff
@@ -1,3 +1,12 @@
 export { useRemixForgeSocketExternal as useRemixForgeSocket } from "./hooks/useRemixForgeSocket";
 export { initClient, initServer } from "./init/project";
+
+export interface Tab {
+  name: string;
+  icon: JSX.Element;
+  id: string;
+  component: JSX.Element;
+  requiresForge: boolean;
+  hideTimeline: boolean;
+}
```
I don't mind the fix but I think the root index file should only export the types, would this work if you did:
```jsx import type { Tab } from ".." export type { Tab }```
or even better:
```jsx import type { Tab } from ".." export type { Tab as RemixDevToolsPluginReturn }```

AlemTuzlak replied:
also this is indicative of the fact that you might be running tsc on your node_modules which you shouldn't do, you should for sure add it to the ignored files in your tsconfig or add the skipLibCheck option

Pickra replied:
I'll check your solution.

In general, I agree about ignoring `node_modules` for typecheck. I'm using prisma which [generates types to node_modules](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/generating-prisma-client) by default. It allows consumers to specify generated output, but that requires more work to consume Remix-Dev-Tools and some people are not gona ignore typecheck on `node_modules `

Pickra replied:
I've updated the change, I think it's what you were asking for.

Pickra replied:
hey @AlemTuzlak , are you cool w/this or do i need to change something?

AlemTuzlak replied:
> hey @AlemTuzlak , are you cool w/this or do i need to change something?

I am so sorry I missed the update you made just saw the notification. Will check asap

AlemTuzlak replied:
> I've updated the change, I think it's what you were asking for.

Mind checking if the latest release solved it?

Pickra replied:
hey @AlemTuzlak,
i updated to v2.1.1, getting a similar error cuz you're importing  `Tab` from a directory that the `dist/` doesn't have. 
I've added a new commit that fixes it.

Pickra commented:
hey @AlemTuzlak,
i updated to v2.1.1, getting a similar error cuz you're importing  `Tab` from a directory that the `dist/` doesn't have. I've added a new commit that works w/the last commit to fix.

## remix dev tools dist in my repo
<img width="853" alt="Screenshot 2023-08-19 at 8 04 08 PM" src="https://github.com/Code-Forge-Net/Remix-Dev-Tools/assets/5226549/ecff2e2b-9dbd-4e3c-99cc-c864022757b5">

AlemTuzlak commented:
> hey @AlemTuzlak,
> i updated to v2.1.1, getting a similar error cuz you're importing `Tab` from a directory that the `dist/` doesn't have. I've added a new commit that works w/the last commit.
> 
> I get that you wana keep the `Tab` type w/the other tab biz but I'm not sure what an alternative solution looks like.
> 
> ## remix dev tools dist in my repo's `node_modules`
> <img width="1229" alt="Screenshot 2023-08-19 at 8 19 46 PM" src="https://github.com/Code-Forge-Net/Remix-Dev-Tools/assets/5226549/1c07bd3e-6a8b-41eb-9701-86474c3545cb">
> 
> <img width="1209" alt="Screenshot 2023-08-19 at 8 29 00 PM" src="https://github.com/Code-Forge-Net/Remix-Dev-Tools/assets/5226549/12033121-c81d-4e9c-8718-6d54f152e0a9">
> 

Ahh thank you so much for this, now I fully understand what the issue is. This is not the way to fix it, I need to change the rollup bundling process to include the tabs types.

Pickra commented:
@AlemTuzlak , I considered changing the rollupconfig exclude/include, but that includes more in the `dist/` and I figured you didn't want that.

AlemTuzlak commented:
> @AlemTuzlak , I considered changing the rollupconfig exclude/include, but that makes  the `dist/` bigger and I figured you didn't want that.
> 
> I'm happy to make that change.

I created a new release, can you check now, had to fix something else so added that too

AlemTuzlak commented:
> I'm getting a different error @AlemTuzlak . leme know if you want me to fix in a different PR. I'm closing this 1.
> 
> <img width="1663" alt="Screenshot 2023-08-21 at 12 33 09 PM" src="https://github.com/Code-Forge-Net/Remix-Dev-Tools/assets/5226549/0ae125f4-6287-4f8f-ace8-2e47b55eb401">
> <img width="1790" alt="Screenshot 2023-08-21 at 12 33 34 PM" src="https://github.com/Code-Forge-Net/Remix-Dev-Tools/assets/5226549/083c6310-4f93-4a7b-94a3-cb0c2ad928b4">
> <img width="1806" alt="Screenshot 2023-08-21 at 12 33 57 PM" src="https://github.com/Code-Forge-Net/Remix-Dev-Tools/assets/5226549/35216694-3e38-4738-a56c-26b300186125">
> 

Yes please!
